pkg/build/ignore/docker/dockign.go

```diff
@@ -0,0 +1,116 @@
+package docker
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DOCKIGN struct {
+	Config *api.Config
+}
+
+func New(req *api.Config) *DOCKIGN {
+	d := &DOCKIGN{
+		Config: req,
+	}
+	return d
+}
+
+func (b *DOCKIGN) Ignore(config *api.Config) {
```

Comment:
So I did a little refactoring and created a "get list of files to ignore method".
However, I expect a disconnect with unit test, as I am in fact temporarily creating files and cleaning them up (where as you pointed to avoiding having stuff on the filesystem)
I of course have a rationale for why actually interfacing with the file system is important here instead of the "fake" pattern (for lack of a better title).  In particular, vetting that the filespec.Glob call is returning what we want I believe is vital.  Faking out other pieces, but avoiding the use of Glob, seems insufficient, but perhaps I still need the veil pulled from my eyes here.

If possible let's stare at the screen together tomorrow once you've had a once over.